### ISSUE TITLE
Error: remove-statements failing for SQS

### ISSUE THREAD

**Issue by hav3wb** [capitalone/cloud-custodian#1816]
Appears to only fail when removing the last statement from an SQS resources. Presumably mirrors issue #1779 for SNS (receiving the same error). Have not checked other resources that use a variation of this action (i.e. lambda, s3, etc), but might be worth checking if others that inherit from the`RemovePolicyBase` class in Actions.py have the same issue.

**Comment by kapilt** [comment]
what's the traceback? Exception?

**Comment by hav3wb** [comment]
@kapilt I get the following error:
```
File "/Users/hjq382/fork_custodian/cloud-custodian/.tox/py27/lib/python2.7/site-packages/c7n/resources/sqs.py", line 142, in process
    results += filter(None, [self.process_resource(client, r)])
  File "/Users/hjq382/fork_custodian/cloud-custodian/.tox/py27/lib/python2.7/site-packages/c7n/resources/sqs.py", line 163, in process_resource
    'Policy':json.dumps(p)
  File "/usr/local/lib/python2.7/site-packages/botocore/client.py", line 312, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/usr/local/lib/python2.7/site-packages/botocore/client.py", line 605, in _make_api_call
    raise error_class(parsed_response, operation_name)
ClientError: An error occurred (InvalidAttributeValue) when calling the SetQueueAttributes operation: Invalid value for the parameter Policy.
```
This appears to mirror the issue described by issue #1779. It only appears to fail when removing the last (or all) permission(s) from an SQS queue. I opened a PR that simply switched to remove_permission instead of set_queue that appears to work for both.

**Comment by hav3wb** [comment]
Fixed by PR #1838
